src/picongpu/include/particles/synchrotronPhotons/PhotonCreator.hpp

```diff
@@ -0,0 +1,289 @@
+    PMACC_ALIGN(cachedB, DataBox<SharedBox<ValueType_B, typename BlockArea::FullSuperCellSize,0> >);
+
+    PMACC_ALIGN(curF_1, SynchrotronFunctions::SyncFuncCursor);
+    PMACC_ALIGN(curF_2, SynchrotronFunctions::SyncFuncCursor);
+
+    /* random number generator for Monte Carlo */
+    typedef ionization::RandomNrForMonteCarlo<T_ElectronSpecies> RandomGen;
+    /* \todo fix: cannot PMACC_ALIGN() because it seems to be too large */
+    RandomGen randomGen;
+
+    float3_X photon_mom;
+
+public:
+    /* host constructor initializing member : random number generator */
+    PhotonCreator(
+        const SynchrotronFunctions::SyncFuncCursor& curF_1,
+        const SynchrotronFunctions::SyncFuncCursor& curF_2,
+        const uint32_t currentStep)
+            : curF_1(curF_1), curF_2(curF_2), randomGen(currentStep)
+    {
+        DataConnector &dc = Environment<>::get().DataConnector();
+        /* initialize pointers on host-side E-(B-)field databoxes */
+        FieldE* fieldE = &(dc.getData<FieldE > (FieldE::getName(), true));
+        FieldB* fieldB = &(dc.getData<FieldB > (FieldB::getName(), true));
+        /* initialize device-side E-(B-)field databoxes */
+        eBox = fieldE->getDeviceDataBox();
+        bBox = fieldB->getDeviceDataBox();
+    }
+
+    /** Initialization function on device
+     *
+     * \brief Cache EM-fields on device
+     *         and initialize possible prerequisites for ionization, like e.g. random number generator.
+     *
+     * This function will be called inline on the device which must happen BEFORE threads diverge
+     * during loop execution. The reason for this is the `__syncthreads()` call which is necessary after
+     * initializing the E-/B-field shared boxes in shared memory.
+     */
+    DINLINE void init(const DataSpace<simDim>& blockCell, const int& linearThreadIdx, const DataSpace<simDim>& localCellOffset)
+    {
+        /* caching of E and B fields */
+        cachedB = CachedBox::create < 0, ValueType_B > (BlockArea());
+        cachedE = CachedBox::create < 1, ValueType_E > (BlockArea());
+
+        /* instance of nvidia assignment operator */
+        nvidia::functors::Assign assign;
+        /* copy fields from global to shared */
+        PMACC_AUTO(fieldBBlock, bBox.shift(blockCell));
+        ThreadCollective<BlockArea> collective(linearThreadIdx);
+        collective(
+                  assign,
+                  cachedB,
+                  fieldBBlock
+                  );
+        /* copy fields from global to shared */
+        PMACC_AUTO(fieldEBlock, eBox.shift(blockCell));
+        collective(
+                  assign,
+                  cachedE,
+                  fieldEBlock
+                  );
+
+        /* wait for shared memory to be initialized */
+        __syncthreads();
+
+        /* initialize random number generator with the local cell index in the simulation*/
+        this->randomGen.init(localCellOffset);
```

Comment:
Is this the original random number generator used in PIConGPU?
Would it make sense to use #1236 (by @Flamefire) instead?
The `seed` in `init(seed)` is for all runs the same. Could you add a dependency on [this seed](https://github.com/ComputationalRadiationPhysics/picongpu/blob/5f82fc19bb78ef0bb6ebe42db44a628454be5c94/src/picongpu/include/simulation_defines/param/seed.param#L40) or [this](https://github.com/ComputationalRadiationPhysics/picongpu/blob/5f82fc19bb78ef0bb6ebe42db44a628454be5c94/src/picongpu/include/simulation_defines/param/seed.param#L47).